### ISSUE TITLE
Add support for edge runtimes

### ISSUE THREAD

**Issue by slowbaker** [lostfictions/znv#2]
Znv causes an error if it's run "on the edge" (for example on [Vecel's Edge Runtime](https://vercel.com/docs/concepts/functions/edge-functions)).

```
error - Error: The edge runtime does not support Node.js 'tty' module.
Learn More: https://nextjs.org/docs/messages/node-module-in-edge-runtime
```

The `tty` module is used by `colorette`.

**Comment by lostfictions** [comment]
Thanks for the report! I'm not using any Node-incompatible runtimes and don't have plans to do so... that said, `colorette` is an extremely small module, so it might be possible to inline it and adapt it to unblock this use-case.

On the other hand, `colorette` already does feature detection and can run fine in non-Node environments like the browser via polyfill. (For example, Webpack 4 polyfills `tty` with a simple module whose [`isatty()`](https://nodejs.org/api/tty.html#ttyisattyfd) function returns `false`; the same polyfill can be [manually included for Webpack 5+](https://gist.github.com/ef4/d2cf5672a93cf241fd47c020b9b3066a).)

Do you know if conditionally requiring `tty` (by wrapping it in a try-catch, say) would fix it? It's a little annoying to do this sort of thing when using TypeScript and it's not really forward-compatible with ES Modules, but I don't mind adding it.

**Comment by dctalbot** [comment]
@lostfictions FWIW I think supporting browsers would be a huge value-add for this library, especially given that `zod` runs in browsers. This issue is blocking adoption of `znv` in my org, so we are having to re-implement the preprocessors that are maintained in this repo. Are you open to accepting PRs for this work?

**Comment by dctalbot** [comment]
To give a specific example of how this would be used: webpack plumbs environment variables into app code using `Webpack.DefinePlugin`. Then in the app code, you would just use `znv.parse(process.env)` to get typed environment variables while developing. You could use the same `znv.parse(process.env)` as a build step on the server and exit the process if there are any errors.

**Comment by dctalbot** [comment]
Nice, I agree supporting ES modules is more important. Thanks for the note of caution - it's funny you mention that because I had the same realization/concern when working with this code recently. create-react-app replaces `process.env` (see [here](https://github.com/facebook/create-react-app/blob/main/packages/react-scripts/config/env.js)), and this is the code my org inherited. It's unfortunately one of many dubious decisions CRA makes despite its widespread adoption.

Instead of clobbering `process.env`, I suppose one could use a new global e.g.
```ts
  const stringified = {
    'MY_APP_ENV_CONFIG': Object.keys(raw).reduce((env, key) => {
      env[key] = JSON.stringify(raw[key]);
      return env;
    }, {}),
  };
```

In app code:
```ts
const env = znv.parse(MY_APP_ENV_CONFIG)
```

On server:

```ts
const env = znv.parse(process.env)
```

IMO supporting other runtimes is still worthwhile, but agreed that any examples in documentation should steer clear of this pitfall.

**Comment by BowlingX** [comment]
@slowbaker Could you try the latest version of `bowlingx-znv` (`0.4.5`) ? I added a check there for the nextjs runtime.
I will create a new PR for that when the ES modules are merged.

**Comment by lostfictions** [comment]
I've just opened #15 which should add edge runtime support via a separate entrypoint -- could folks here who requested this feature take a look and see if it works for you?

I've published it to npm under the `next` tag, so you can `npm i`/`pnpm add`/`yarn add` `znv@next` to try it out.
